src/VTUFiles.f90

```diff
@@ -1,643 +1,497 @@
+      CHARACTER(LEN=*),OPTIONAL,INTENT(IN) :: pad
+      INTEGER(SIK),OPTIONAL,INTENT(IN) :: recl
+      !
+      IF(.NOT.fileobj%isInit()) THEN
+        IF(PRESENT(access)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "ACCESS" is being ignored. Value is "SEQUENTIAL".')
+        IF(PRESENT(form)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "FORM" is being ignored. Value is "FORMATTED".')
+        IF(PRESENT(action)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "ACTION" is being ignored. Value is "WRITE".')
+        IF(PRESENT(pad)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "PAD" is being ignored. Value is "YES".')
+        IF(PRESENT(position)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "POSITION" is being ignored. Value is "APPEND".')
+        IF(PRESENT(recl)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "RECL" is being ignored. File is "SEQUENTIAL".')
+        !
+        !Initialize the file, only support ascii for now
+        CALL init_fortran_file(fileobj,unit,file,'REPLACE','SEQUENTIAL', &
+          'FORMATTED','APPEND','WRITE')
+        !
+        !Open the file
+        CALL fileobj%fopen()
+        !
+        !Handle status argument, which is unused
+        IF(PRESENT(status)) THEN
+          CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+            ' - File header not used for VTU files!')
+        ENDIF
+        !
+        !Write header to file, title and format information to file
+        IF(fileobj%isOpen()) THEN
+          WRITE(fileobj%getUnitNo(),'(a)') '<?xml version="1.0"?>'
+          WRITE(fileobj%getUnitNo(),'(a)') '<VTKFile type='// &
+          '"UnstructuredGrid" version="0.1" byte_order="LittleEndian">'
+          !Change "new" status so that if file is closed and re-opened
+          !during execution it will be appended file instead of replacing it.
+          fileobj%newstat=.FALSE.
+        ENDIF
+      ELSE
+        CALL fileobj%e%raiseError(modName//'::'//myName// &
+          ' - VTU File is already initialized!')
+      ENDIF
+    ENDSUBROUTINE init_VTUXMLFileType
+!
+!-------------------------------------------------------------------------------
+!> @brief Writes a VTU mesh to VTU file
+!> @param myVTKFile the VTK file to write the data to
+!> @param vtkMesh the VTK mesh to write in the file
+!> 
+!> This routine supports writing of UNSTRUCTURED_GRIDS. The other types of 
+!> geometry/topology are not yet implemented. The display of this 
+!> data was verified by loading and displaying it with VisIt.
+!>
+    SUBROUTINE writeMesh_VTUXMLFileType(myVTKFile,vtkMesh)
+      CHARACTER(LEN=*),PARAMETER :: myName='writeMesh_VTUXMLFileType'
+      CLASS(VTUXMLFileType),INTENT(INOUT) :: myVTKFile
+      TYPE(VTKMeshType),INTENT(IN) :: vtkMesh
+      CHARACTER(LEN=256) :: aline,sint
```

Comment:
What is the justification for 256 here? Do you anticipate any problems with this?